deploy/eck-operator/templates/configmap.yaml

```diff
@@ -9,7 +9,22 @@ metadata:
 data:
   eck.yaml: |-
     log-verbosity: {{ int .Values.config.logVerbosity }}
-    metrics-port: {{ int .Values.config.metricsPort }}
+    {{- if (and .Values.config.enableSecureMetrics (eq (int .Values.config.metricsPort) 0) (or (eq .Values.config.metricsBindAddress "") (not (hasKey .Values.config "metricsBindAddress")))) }}
+    {{ fail "metricsPort or metricsBindAddress is required when enableSecureMetrics is true" }}
+    {{- end }}
+    {{- if (and .Values.config.enableSecureMetrics (gt (int .Values.config.metricsPort) 0) (or (eq .Values.config.metricsBindAddress "") (not (hasKey .Values.config "metricsBindAddress")))) }}
+    metrics-port: {{ add (int .Values.config.metricsPort) 1 }}
+    {{- else }}
+    metrics-port: {{ int (.Values.config.metricsBindAddress | splitList ":" | last) }}
+    {{- end }}
+    {{- if (and .Values.config.enableSecureMetrics (hasKey .Values.config "metricsBindAddress") (not (eq .Values.config.metricsBindAddress ""))) }}
+    {{- if .Values.config.enableSecureMetrics }}
+    {{ $host := .Values.config.metricsBindAddress | splitList ":" | first }}
+    metrics-bind-address: {{ $host }}:{{ add (int (.Values.config.metricsBindAddress | splitList ":" | last)) 1 }}
+    {{- else }}
+    metrics-bind-address: {{ int .Values.config.metricsBindAddress | splitList ":" | last }}
+    {{- end }}
+    {{- end }} 
     container-registry: {{ .Values.config.containerRegistry }}
```

Comment:
Having to support both `metricsBindAddress` and `metricsPort` is making this logic overly complex. I initially tried to to add something to `_helpers.tpl`, but since that only provides strings, this really didn't make things much cleaner.

Suggestions to simplify this would be greatly appreciated.

Reply:
This simplified things greatly. Thanks @pebrc